resources/analysisTools/roddyTools/wrapInScript.sh

```diff
@@ -48,20 +57,44 @@ declare_xg() {
     eval "export $varName=\"$value\""
 }
 
+getEnvironmentScriptPath() {
+    local varName="${1:?No variable name given}"
+    if (echo "${!varName}" | grep -P '^\${'); then
+        local scriptPath="${!varName}"
+    else
+        local tmp
+        tmp=$(createToolVariableName "${!varName}")
+        local transformedName="$tmp"
+        local scriptPath="${!transformedName}"
+    fi
+    if [[ -z "$scriptPath" ]]; then
+        echo "Requested environment script variable '$varName' does not point to a value" > /dev/stderr
+        exit 200
+    fi
+    echo "$scriptPath"
+}
+
+warnEnvironmentScriptOverride() {
+    if [[ -n "$ENVIRONMENT_SCRIPT" ]]; then
+        echo "ENVIRONMENT_SCRIPT variable is set externally (e.g. in the XML) to '$TOOL_ENVIRONMENT'. It will be reset." > /dev/stderr
+    fi
+}
+
 # Basic modules / environment support
 # Load the environment script (source), if it is defined. If the file is defined but the file not accessible exit with
 # code 200. Additionally, expose the used environment script path as ENVIRONMENT_SCRIPT variable to the wrapped script.
 runEnvironmentSetupScript() {
     local envScriptVar="${TOOL_ID}EnvironmentScript"
     if [[ -n "${!envScriptVar}" ]]; then
-        # declare -gx ENVIRONMENT_SCRIPT="${!envScriptVar}"
-        declare_xg ENVIRONMENT_SCRIPT "${!envScriptVar}"
+	    warnEnvironmentScriptOverride
```

Comment:
That's basically the same code as for the else branch. You could reduce it to two method calls and implement a new method. But it is ok like it is.

Reply:
I'll change it.